core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1510,10 +1510,21 @@ class UnifiedLog(@volatile var logStartOffset: Long,
           }
         }
         localLog.checkIfMemoryMappedBufferClosed()
-        // remove the segments for lookups
-        localLog.removeAndDeleteSegments(segmentsToDelete, asyncDelete = true, reason)
-        deleteProducerSnapshots(deletable, asyncDelete = true)
-        incrementStartOffset(localLog.segments.firstSegmentBaseOffset.getAsLong, LogStartOffsetIncrementReason.SegmentDeletion)
+        if (remoteLogEnabled()) {
+          /**
+           * See @link{https://issues.apache.org/jira/browse/KAFKA-16073} for the background of this fix
+           */
+          val newLocalLogStartOffset = localLog.segments.higherSegment(deletable.last.baseOffset()).get().baseOffset()
```

Comment:
Instead of using `deletable`, can we use `segmentsToDelete` to find the new-local-log-start-offset? 

```java
val newLocalLogStartOffset = localLog.segments.higherSegment(segmentsToDelete.last.baseOffset()).get().baseOffset()
```

Reply:
We should handle the case where we are deleting the only segment which is active segment.

Reply:
Seems the active-segment deletion case is already handled, when the topic is enabled with remote-log-storage, then the below condition will always be false:

```
if (localLog.segments.numberOfSegments == numToDelete) 
```

Reply:
I would suggest still making the change to use `segmentsToDelete` instead of `deletable` as that is more accurate and covers if we change the logic to generate deletable segments in tiered-storage path in future.